matrix/matrix.py

```diff
@@ -0,0 +1,198 @@
+    """
+    __slots__ = '_matrix'
+
+    def __init__(self, *lines):
+        if not self._is_valid_data(*lines):
+            raise ValueError('Data is not valid')
+
+        super(Matrix, self).__setattr__('_matrix', tuple(lines))
+
+    def __setattr__(self, key, value):
+        """Defines that Matrix attributes are immutable."""
+        raise AttributeError(f'{self.__class__} instance cannot be modified')
+
+    @staticmethod
+    def _is_valid_data(*data: tuple) -> bool:
+        """Checks if the data given is valid. Valid data is defined if it
+        contains tuples of the same length (Matrix only supports n * n) and
+        their value is an integer."""
+        # check if each line is a tuple.
+        if not all([isinstance(line_data, tuple) for line_data in data]):
+            return False
+
+        # check if each item in each line is an int.
+        if not all([any([isinstance(item, int), isinstance(item, float)]) for
+                    line_data in data for item in line_data]):
+            return False
+
+        # checks if the given data is of a n * n matrix.
+        if not all([len(line) == len(data) for line in data]):
+            return False
+
+        return True
+
+    @classmethod
+    def unity(cls, number_lines: int):
+        """Generates a matrix with a main diagonal of 1's, everything else
+        is 0's.
+        """
+        tuples_in_chars = [
+            list(str('1' + ('0' * i)).zfill(number_lines)[::-1]) for i in
+            range(number_lines)]
+
+        return cls(*tuple(
+            [tuple(map(int, tuple_chars)) for tuple_chars in tuples_in_chars]))
+
+    @classmethod
+    def ones(cls, number_lines):
+        """Generates a matrix full of 1's."""
+        return cls(*tuple(
+            [tuple(repeat(1, number_lines)) for _ in range(number_lines)]))
+
+    def __repr__(self):
+        """Returns a string presentation of the matrix instance."""
+        return f"({', '.join([str(line) for line in self._matrix])})"
+
+    @property
+    def tuples(self):
+        """Returns the matrix instance as a tuple of tuples."""
+        return self._matrix
+
+    @staticmethod
+    def _bool_operation(operation, data) -> bool:
+        """Checks boolean operation output.
+
+        Args:
+            operation (function): the boolean function to operate.
+            data (list): the data of the matrix (list of tuples).
+
+        Returns:
+            If the operation is 'equal', returns true if all items are equal
+            to their parallel items in other, else false. If operation is
+            'not equal', return true if one or more items are not equal to
+            their parallel item/s in other, else false.
+        """
+        if operation == operator.eq:
+            return all(
+                [item for line_data in data for item in line_data])
+
+        elif operation == operator.ne:
+            return any(
+                [item for line_data in data for item in line_data])
```

Comment:
Instead of `if-elif` which is considered a code-smell in Python, you could have just called the correct function as follows:

```suggestion
        func = {operator.eq: all, operator.ne: any}.get(operation)
        return func(
                [item for line_data in data for item in line_data])
```